### PR TITLE
Update flutter and dependencies major versions

### PR DESCRIPTION
**Pull request by omar-selo**
Flutter version got bumped up. If we used the latest flutter the code wouldn't work cause some packages we're depending on need updating to be compatible. So I figured I'd upgrade the major version of everything. This ofc involved a few changes to the code.

### CODE REVIEW

File: .github/workflows/test_frontend.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
       - uses: actions/checkout@v3
       - uses: subosito/flutter-action@v2
         with:
-          flutter-version: '3.10.6'
+          flutter-version: '3.13.0'
```
Comment (p-gentili) [inline .github/workflows/test_frontend.yml:17]: Since we're bumping, why not the latest? 3.13.7

File: frontend/Dockerfile [hunk 2]
```diff
@@ -6,7 +6,7 @@ RUN apt-get update
 RUN apt-get install -y curl git wget unzip libgconf-2-4 gdb libstdc++6 libglu1-mesa fonts-droid-fallback lib32stdc++6 python3
 
 RUN git clone https://github.com/flutter/flutter.git /opt/flutter && \
-    cd /opt/flutter && git checkout '3.10.6' && \
+    cd /opt/flutter && git checkout '3.13.0' && \
```
Comment (p-gentili) [inline frontend/Dockerfile:9]: ditto

File: frontend/lib/models/test_execution.dart [hunk 3]
```diff
@@ -57,7 +57,7 @@ enum TestExecutionStatus {
       case inProgress:
         return const Icon(YaruIcons.refresh, size: size);
       case passed:
-        return const Icon(YaruIcons.ok, color: YaruColors.success, size: size);
+        return Icon(YaruIcons.ok, color: YaruColors.light.success, size: size);
```
Comment (p-gentili) [inline frontend/lib/models/test_execution.dart:60]: dumb Q: are we Ok with not checking the theme here?

**Comment by p-gentili** [comment]
Ops, already merged :)

File: frontend/lib/models/test_execution.dart (hunk 3 above)
Comment (omar-selo) [reply frontend/lib/models/test_execution.dart:60]: Both dark and light themes have the same color for success. But mainly, our dashboard doesn't support different themes. i.e. you'll see the same colors irrelevant of what you pick

File: .github/workflows/test_frontend.yml (hunk 1 above)
Comment (omar-selo) [reply .github/workflows/test_frontend.yml:17]: Now I understand what's going on. I'm using flutter snap. I thought a snap refresh would mean I get the latest flutter version. Turns out no, a refresh just refreshes the snap but not flutter itself. To upgrade flutter I'd need to run flutter upgrade. This makes sense as we wouldn't want snap automatically upgrading flutter versions.

**Pull request merged**